packages/liferay-npm-scripts/src/jsp/lintJSP.js

```diff
@@ -72,19 +72,21 @@ function lintJSP(source, onReport, options = {}) {
 				return true;
 			});
 
+			// Ensure trailing newline, matching Prettier's convention.
+			source = output.endsWith('\n') ? output : `${output}\n`;
+
```

Comment:
Moved this up here for consistency with the equivalent code in `lintSCSS`; it seems that the `source` we report via the `onReport` callback should be the same as the string we return from this function.